mixin/lib/civimix-schema@5/src/SqlGenerator.php

```diff
@@ -156,6 +157,22 @@ private function getTableConstraints(array $entity): array {
     return $constraints;
   }
 
+  private function getConstraintName(string $tableName, string $fieldName): string {
+    $maxLength = 63;
+    // If string is too long, start by trimming the "civi" off the beginning of the table so it just starts with "crm_"
+    if (strlen("FK_{$tableName}_$fieldName") > $maxLength) {
+      $tableName = substr($tableName, 4);
+    }
+    // If string is still too long, truncate table name and field name in equal proportion.
+    if (strlen("FK_{$tableName}_$fieldName") > $maxLength) {
+      // Remaining length is 63 - strlen(FK_) minus 1 for the remaining separator
+      $remainingLength = $maxLength - 4;
+      $tableName = substr($tableName, 0, floor($remainingLength / 2));
+      $fieldName = substr($fieldName, 0, $remainingLength - strlen($tableName));
+    }
+    return "FK_{$tableName}_$fieldName";
```

Comment:
This is not guaranteed to be unique at this point. Would it make sense to check and ensure there will be no error?

Reply:
Good point....

FWIW, just confirming the requirement: https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html:

>  The CONSTRAINT symbol value, if defined, must be unique in the database. A duplicate symbol results in an error similar to: ERROR 1005 (HY000): Can't create table 'test.fk1' (errno: 121).